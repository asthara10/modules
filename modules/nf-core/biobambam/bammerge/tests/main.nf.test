nextflow_process {

    name "Test Process BIOBAMBAM_BAMMERGE"
    script "../main.nf"
    config "./nextflow.config"
    process "BIOBAMBAM_BAMMERGE"

    tag "modules"
    tag "modules_nfcore"
    tag "biobambam"
    tag "biobambam/bammerge"

    test("test_biobambam_bammerge_paired") {


        when {
            process {
                """
                
                input[0] = [
        [ id:'test', single_end:false ], // meta map
        [
            file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
            file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_sorted_bam'], checkIfExists: true)
        ]
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_biobambam_bammerge_single") {
        when {
            process {
                """
                        input[0] = [
        [ id:'test', single_end:false ], // meta map
        [
            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
        ]
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
