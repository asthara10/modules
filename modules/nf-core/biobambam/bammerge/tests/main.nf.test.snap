{
    "versions": {
        "content": [
            [
                
            ]
        ],
        "timestamp": "2023-11-24T14:37:33.407428"
    }
}